docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -143,77 +209,6 @@ const finalTheme = createTheme({
 
 If you use the theming approach to customize the components, you'll still be able to override them using the `sx` prop as it has a higher CSS specificity, even if you're using the experimental `sx` syntax within the theme.
 
-## Creating new component variants
```

Comment:
I think it was clearer what it was possible to achieve by having a dedicated section on how to add new variants. How can users add new variants now that variants are only accessible through slots? By adding new variants to the `root` slots?

Reply:
> how can users add new variants now that variants are only accessible through slots? By adding new variants to the root slots?

Yes, that's correct. I wonder if the `### Overrides based on props` is not clear? I feel like the question contains answer in itself "how can users add new variants now that variants are only accessible through slots".

Reply:
> I feel like the question contains answer in itself

It wasn't clear to me because I'm not familiar with the implementation. I think the mental model was clearer when using:

```js
MuiButton: {
  variants: [
    {
      props: { variant: 'dashed' },
      style: { ... }
    }
  ]
}
```

because you're adding a variant to the component, which is how users and tools treat variants.

Adding a variant to a slot is conceptually different from adding it to the component, and I can see how it can cause some confusion and doubts.

```js
MuiButton: {
  styleOverrides: {
    root: {
      variants: [
        {
          props: { variant: 'dashed' },
          style: { ... }
        }
      ]
    }
  }
}
```


Question users might have:

"Are all new variant values declared in all slots available when instantiating the component? Or only the ones added to the `root` slot?"

Reply:
I see. This is Material UI v6 specific, users will need to learn that a component is composed of 1 or more slots. They should be customizing the styles of component's slot (a mapping to DOM element).

The `props` they specified in each of variant will match with the component props (all slots receive the same props).

![image](https://github.com/mui/material-ui/assets/18292247/373626b2-249a-424d-9765-af14bcb58af0)

I listed the benefits of targeting slot than the component in https://github.com/mui/material-ui/issues/30412#issuecomment-1894892080.